### PR TITLE
Make vartime suites work

### PR DESCRIPTION
**Pull request by jeffallen**
When compiled with "-tags vartime", the suite for cothority is P256.

When the integration tests are run with env var TAGS set to "-tags vartime", the suite for cothority is P256.

When the same env variable is set, run_conode.sh compiles conode with vartime, and the default suite is P256. If you set up a new node, it's suite will be P256. 

The suite is named in the toml files, but if it is not named it defaults to Ed25519, so old files can be used unchanged.

There are still lots of parts of the system that do not allow you to configure the suite at runtime (they use cothority.Suite instead). But at least with these changes, we've been able to identify and fix incorrect assumptions about key lengths.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 language: go
 
 install:
+  - go get github.com/dedis/onet
+  - ( cd $GOPATH/src/github.com/dedis/onet ; git checkout p256 )
```
Comment (ineiti) [inline .travis.yml]: That can't be right in master...

File: bftcosi/bftcosi.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 	"github.com/dedis/onet/log"
 )
 
-const timeout = 1
+const defaultTimeout = 1 * time.Second
```
Comment (ineiti) [inline bftcosi/bftcosi.go]: This should be in another PR, no?

File: bftcosi/packets.go [hunk 3]
```diff
@@ -70,6 +71,10 @@ func (bs *BFTSignature) Verify(s network.Suite, publics []kyber.Point) error {
 	origCommit := s.Point()
 	pointLen := s.PointLen()
 	sigLen := pointLen + s.ScalarLen()
+
+	if len(bs.Sig) < sigLen {
```
Comment (ineiti) [inline bftcosi/packets.go]: Shouldn't this be `!=`?

File: identity/api_test.go [hunk 4]
```diff
@@ -75,7 +83,8 @@ func TestIdentity_StoreKeys(t *testing.T) {
 	signature := make(chan []byte)
 	c := node.(*cosi.CoSi)
 	c.RegisterSignatureHook(func(sig []byte) {
-		signature <- sig[:64]
+		log.LLvl3("sig", len(sig))
```
Comment (ineiti) [inline identity/api_test.go]: `s/LLvl/Lvl/`

File: messaging/propagate.go [hunk 5]
```diff
@@ -17,7 +17,7 @@ func init() {
 }
 
 // How long to wait before timing out on waiting for the time-out.
-const initialWait = 100000
+const initialWait = 100000 * time.Millisecond
```
Comment (ineiti) [inline messaging/propagate.go]: can we have the timeouts in another PR, please?

File: skipchain/msgs.go [hunk 6]
```diff
@@ -58,8 +58,8 @@ func init() {
 		&ProtoExtendSignature{},
 		&ProtoExtendRoster{},
 		&ProtoExtendRosterReply{},
-		&ProtoGetUpdate{},
-		&ProtoBlockReply{},
+		&ProtoGetBlocks{},
```
Comment (ineiti) [inline skipchain/msgs.go]: Also another PR for the `GetBlocks`

**Comment by jeffallen** [comment]
This PR is totally screwed up, replacing with several that are correct.

File: bftcosi/packets.go (hunk 3 above)
Comment (jeffallen) [reply bftcosi/packets.go]: No, there's a variable length mask appended onto the end (cosi.go, line 204).
